pkg/cleanup/types/built_in.go

```diff
@@ -133,36 +134,52 @@ func scServiceInstanceCleanup(spec, actual *unstructured.Unstructured) (runtime.
 		return nil, err
 	}
 
-	if instanceSpec.Spec.ClusterServiceClassName != "" &&
-		instanceSpec.Spec.ClusterServiceClassName != instanceActual.Spec.ClusterServiceClassName {
-		return nil, errors.New("clusterServiceClassName has changed when it should be immutable")
-	}
+	instanceSpec.ObjectMeta.Finalizers = instanceActual.ObjectMeta.Finalizers
 
-	if instanceSpec.Spec.ClusterServicePlanName != "" &&
-		instanceSpec.Spec.ClusterServicePlanName != instanceActual.Spec.ClusterServicePlanName {
-		return nil, errors.New("clusterServicePlanName has changed when it should be immutable")
-	}
+	setEmptyFieldsFromActual(&instanceSpec.Spec, &instanceActual.Spec, []string{
+		"ClusterServiceClassName",
+		"ClusterServiceClassRef",
+		"ClusterServiceClassExternalName",
+		"ClusterServiceClassExternalID",
 
-	if instanceActual.Spec.ClusterServiceClassExternalName == instanceSpec.Spec.ClusterServiceClassExternalName {
```

Comment:
It's not clear to me why this logic is different (and why is uses ExternalName rather than ExternalID).

At any rate, I don't think it's necessary - would prefer to keep strange service catalog specific knowledge out of here.